actions/poll_status.py

```diff
@@ -25,17 +29,50 @@ def query(self, url, verify_ssl_cert):
             resp = requests.get(url, verify=verify_ssl_cert)
             return resp
         except RequestException as err:
-            current_time = datetime.datetime.now()
-            self.logger.warning(
-                "{0} -- {1} - an error was encountered: {2}".format(current_time, url, err))
-            return None, None
+            self.logger.warning("An error was encountered when "
+                                "querying url: {0},  {1}".format(url, err))
+            return None
+
+    def post_to_endpoint(self, endpoint, body, irma_mode, verify_ssl_cert):
 
-    def run(self, url, sleep, ignore_result, verify_ssl_cert, max_retries=3):
+        def _rewrite_link(link):
+            endpoint_parsed = urlparse(endpoint)
+            first_part_of_path = endpoint_parsed.path.split('/')[1]
```

Comment:
We seem to define the destination URL upstream (in `group_vars/arteria-master/main.yml`) on the following format 

```irma_checksum_base_url: https://irma1.uppmax.uu.se:4444/arteria_checksum/api/1.0```

If we would change the way we define these URLs, by e.g. splitting out the host part and the path part, then we perhaps we might run the risk of having your `endpoint_parsed.path.split('/')[1]` returning an empty string if the full URL is concated together upstreams somewhere from e.g. `https://irma1.uppmax.uu.se:4444/` + `/arteria_checksum/api/1.0` into `https://irma1.uppmax.uu.se:4444//arteria_checksum/api/1.0`. 

Although I guess it is not very likely that we would end up writing the URLs in that way (compare to file paths though, where it is much more likely that you have a trailing `/` to indicate that it is a folder). And perhaps `urlparse()` also fixes this for you. So it is probably just a moot point anyway.